TamperMonkey/BIT-乐学-修改侧边栏课程-GUI.user.js

```diff
@@ -240,20 +220,21 @@ span[contenteditable='true']{
     const xhr = new XMLHttpRequest()
     xhr.onreadystatechange = function () {
       if (xhr.readyState === 4 && xhr.status === 200) {
-        shown_courses = []
-        hidden_courses = []
-        const a = new DOMParser().parseFromString(xhr.responseText, 'text/html').querySelectorAll("a[href*='lexue.bit.edu.cn/user/view.php']")
-        for (let i = 0; i < a.length; i++) {
-          shown_courses.push([a[i].href.match(/(?<=course=)\w+/)[0], a[i].innerText])
+        const courses = JSON.parse(xhr.responseText)[0].data.courses
+        courses.sort((a, b) => a.enddate - b.enddate)
+        for (let i = 0; i < courses.length; i++) {
+          if (!hidden_courses.filter(hidden_course => parseInt(hidden_course[0]) === courses[i].id).length && !shown_courses.filter(shown_course => parseInt(shown_course[0]) === courses[i].id).length) {
+            hidden_courses.unshift([courses[i].id, `<b style="color:red">${courses[i].fullname}</b>`, 'graduation-cap'])
+          }
         }
         openPopup()
       }
     }
-    xhr.open('GET', document.querySelector("[href*='lexue.bit.edu.cn/user/profile.php?id=']").href + '&showallcourses=1', true)
-    xhr.send()
+    xhr.open('POST', `https://lexue.bit.edu.cn/lib/ajax/service.php?sesskey=${M.cfg.sesskey}&info=core_course_get_enrolled_courses_by_timeline_classification`, true)
```

Comment:
https://results.pre-commit.ci/run/github/330974151/1728363997.VtGk4APBTq2u1uiP1SuJAw 说这个`M`没有定义，是否有误？

Reply:
M是乐学的全局变量，经测试能够正常使用

Reply:
原来如此；那建议在开头标一下。

```suggestion
  // 乐学提供
  /* global M */
```

https://eslint.org/docs/latest/use/configure/language-options#using-configuration-comments